packages/demo/src/components/Login/index.js

```diff
@@ -1,35 +1,123 @@
 import React from 'react'
-import {connect} from 'cerebral/react'
+import { connect } from 'cerebral/react'
 import translations from '../../computed/translations'
+import LangSelector from '../LangSelector'
+import Input from './Input'
 
 export default connect(
   {
-    t: translations
+    t: translations,
+    email: 'user.$signIn.email',
+    password: 'user.$signIn.password',
+    validationErrors: 'user.$signIn.validationErrors',
+    error: 'user.$signIn.error'
   },
-  function Login ({t}) {
+  {
+    fieldChanged: 'user.fieldChanged',
+    signInClicked: 'user.signInClicked',
+    createUserClicked: 'user.createUserClicked'
+  },
+  function Login ({
+    t,
+    // state
+    email,
+    password,
+    validationErrors,
+    error,
+    // signals
+    fieldChanged,
+    signInClicked,
+    createUserClicked
+  }) {
+    let emailValidationMessage = ''
+    let passwordValidationMessage = ''
+
+    function validate () {
+      // validate email
+      let hasValidationError = validationErrors && validationErrors.EMAIL_EMPTY
+      if (hasValidationError) {
+        emailValidationMessage = t.loginValidationEmailRequired
+      } else if (error) {
+        let translation = t.loginErrors[error.code]
+        emailValidationMessage = translation || error.message
+      }
+
+      // validate password
+      hasValidationError = validationErrors && validationErrors.PASSWORD_EMPTY
+      if (hasValidationError) {
+        passwordValidationMessage = t.loginValidationPasswordRequired
+      }
+    }
+
+    validate()
```

Comment:
I would move the validationErrors inside the Input like this:

```jsx
<Input fieldType='email' errors={validationErrors.email} ... />
```

With validationErrors being an object (always defined) like this:

```js
validationErrors = {
   email: EMAIL_EMPTY
}
```

Then display and translation of error message is all handled by the Input field...